libs/compute_api/src/spec.rs

```diff
@@ -0,0 +1,302 @@
+//! `ComputeSpec` represents the contents of the spec.json file.
+//!
+//! The spec.json file is used to pass information to 'compute_ctl'. It contains
+//! all the information needed to start up the right version of PostgreSQL,
+//! and connect it to the storage nodes.
+use anyhow::anyhow;
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, DisplayFromStr};
+use std::collections::HashMap;
+use std::fmt::Write;
+use std::str::FromStr;
+use utils::id::{TenantId, TimelineId};
+use utils::lsn::Lsn;
+
+use crate::rfc3339_serialize;
+
+/// String type alias representing Postgres identifier and
+/// intended to be used for DB / role names.
+pub type PgIdent = String;
+
+/// Cluster spec or configuration represented as an optional number of
+/// delta operations + final cluster state description.
+#[serde_as]
+#[derive(Clone, Deserialize, Serialize)]
+pub struct ComputeSpecV2 {
+    pub format_version: u64,
+
+    // For debugging purposes only
+    pub project_id: Option<String>,
+    pub endpoint_name: Option<String>,
+    pub operation_uuid: Option<String>,
+
+    /// Timestamp when the compute launch started, for debugging / tracing
+    #[serde(serialize_with = "rfc3339_serialize")]
+    pub startup_timestamp: DateTime<Utc>,
```

Comment:
I don't have really better suggestions right now, but with changes I started to work on these `startup_*` will look a bit odd. Imagine:
- Compute crashed and pod / vm restarted, so compute_ctl goes to the control-plane to get the latest spec. It receives some old timestamp, when it was initially generated / started, but this doesn't make much sense
- We send updated spec to the compute to do reconfiguration, then it's already started

I'd maybe make this optional at least, same as we did with `startup_tracing_context` in your previous PR

Reply:
Maybe we should just remove `startup_timestamp`? The control plane can log or store the timestamp all on its own, we don't really need it for anything in `compute_ctl`.

Reply:
> Maybe we should just remove startup_timestamp? The control plane can log or store the timestamp all on its own, we don't really need it for anything in compute_ctl.

Yeah, that was my first intention too. We also have optional tracing context for exactly the same purpose. And all important steps are logged with timestamp in the console / compute_ctl already